eng/scripts/Test-Packages.ps1

```diff
@@ -30,22 +31,24 @@ if ($PackageInfoPath) {
   | ConvertFrom-Json
 }
 else {
-  $packagesToTest = Get-AllPackagesInRepo
+  . "$PSScriptRoot\Language-Settings.ps1"
+
+  $packagesToTest = Get-AllPackageInfoFromRepo
 }
 
 Write-Host "Testing packages:"
 foreach ($package in $packagesToTest) {
-  Write-Host "  '$($package.Name)'"
+  Write-Host "  '$($package.Name)' in '$($package.DirectoryPath)'"
 }
 
 Write-Host "Setting RUSTFLAGS to '-Dwarnings'"
 $env:RUSTFLAGS = "-Dwarnings"
 
 
 foreach ($package in $packagesToTest) {
-  Push-Location (Join-Path $RepoRoot $package.DirectoryPath)
+  Push-Location ([System.IO.Path]::Combine($RepoRoot, $package.DirectoryPath))
```

Comment:
Were you seeing a difference between Path::Combine and Join-Path here?

Reply:
Nevermind.  I saw your note re: combining absolute paths:
```
> Join-Path  'c:\something\else' 'd:\another\path'
c:\something\else\d:\another\path

> [IO.Path]::Combine('c:\something\else', 'd:\another\path')
d:\another\path
```

Reply:
Yeah. Never use `Join-Path`. It's utter garbage and gets people into trouble so many times. It's worse on non-Windows where you don't get a drive letter in the middle of two absolute paths - just `/`s.